From c48955830d72693c1b3114c0cea357967df1e1b3 Mon Sep 17 00:00:00 2001
From: Huiquan Zhong <huiquan.zhong@intel.com>
Date: Thu, 30 Jul 2015 17:19:04 +0800
Subject: [PATCH] spi/pxa2xx: add dmaengine_resume() in resume callback

Add dmaengine_resume() in dma resume callback.

Also call pxa2xx_spi_dma_resume() in .runtime_resume callabck.

Change-Id: Id1e1f40bd2e27768597faef4df22aad7f8f34799
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-38381
Signed-off-by: Huiquan Zhong <huiquan.zhong@intel.com>
---
 drivers/spi/spi-pxa2xx-dma.c | 4 ++++
 drivers/spi/spi-pxa2xx.c     | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/drivers/spi/spi-pxa2xx-dma.c b/drivers/spi/spi-pxa2xx-dma.c
index 6fc2bc1cb9fa..cba703498368 100644
--- a/drivers/spi/spi-pxa2xx-dma.c
+++ b/drivers/spi/spi-pxa2xx-dma.c
@@ -355,6 +355,10 @@ void pxa2xx_spi_dma_release(struct driver_data *drv_data)
 
 void pxa2xx_spi_dma_resume(struct driver_data *drv_data)
 {
+	if (drv_data->rx_chan)
+		dmaengine_resume(drv_data->rx_chan);
+	if (drv_data->tx_chan)
+		dmaengine_resume(drv_data->tx_chan);
 }
 
 int pxa2xx_spi_set_dma_burst_and_threshold(struct chip_data *chip,
diff --git a/drivers/spi/spi-pxa2xx.c b/drivers/spi/spi-pxa2xx.c
index 07424a86b257..226b1e6e6bcf 100644
--- a/drivers/spi/spi-pxa2xx.c
+++ b/drivers/spi/spi-pxa2xx.c
@@ -1405,6 +1405,8 @@ static int pxa2xx_spi_runtime_resume(struct device *dev)
 {
 	struct driver_data *drv_data = dev_get_drvdata(dev);
 
+	pxa2xx_spi_dma_resume(drv_data);
+
 	clk_prepare_enable(drv_data->ssp->clk);
 
 	/* Init LPSS private register bits */
-- 
1.9.1

